### PR TITLE
gradle: 5.6.4 -> 6.5.1

### PR DESCRIPTION
**Pull request by ludovicc**
###### Motivation for this change

Upgrade Gradle to latest version 6.5.1
Add gradle_6 to the top level packages.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
@ludovicc thanks for opening your first pr to nixpkgs :)

File: pkgs/development/tools/build-managers/gradle/default.nix [hunk 1]
```diff
@@ -51,7 +51,17 @@ rec {
...
 
-  gradle_latest = gradle_5_6;
+  gradle_latest = gradle_6_5;
+
+  gradle_6_5 = gradleGen rec {
+    name = "gradle-6.5.1";
+    nativeVersion = "0.22";
+
+    src = fetchurl {
+      url = "http://services.gradle.org/distributions/${name}-bin.zip";
```
Comment (jonringer) [inline pkgs/development/tools/build-managers/gradle/default.nix:61]: I get a 403:
```
builder for '/nix/store/v34wfmafk54chdkascgsycwmsdnw1ypc-gradle-6.5.1-bin.zip.drv' failed with exit code 1; last 7 log lines:

  trying http://services.gradle.org/distributions/gradle-6.5.1-bin.zip
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  curl: (22) The requested URL returned error: 403 Forbidden
  error: cannot download gradle-6.5.1-bin.zip from any mirror
```

**Comment by ludovicc** [comment]
@jonringer Thanks for the welcome and the review. It looks like you had some kind of transient error while downloading the zip, http://services.gradle.org/distributions/gradle-6.5.1-bin.zip works both from browser and using nix-review on this pr.

File: pkgs/development/tools/build-managers/gradle/default.nix (hunk 1 above)
Comment (danieldk) [inline pkgs/development/tools/build-managers/gradle/default.nix:61]:
```suggestion
      url = "https://services.gradle.org/distributions/${name}-bin.zip";
```

I get a 403 for that URL as well, but if I change this to use https it works:

```
❯ nix-prefetch-url http://services.gradle.org/distributions/gradle-6.5.1-bin.zip 
[0.0 MiB DL]
error: unable to download 'http://services.gradle.org/distributions/gradle-6.5.1-bin.zip': HTTP error 403
❯ nix-prefetch-url https://services.gradle.org/distributions/gradle-6.5.1-bin.zip
[97.6 MiB DL]
path is '/nix/store/fk10bkypfhjnsgsc2b2lyxxpnqpghs31-gradle-6.5.1-bin.zip'
0jmmipjh4fbsn92zpifa5cqg5ws2a4ha0s4jzqhrg4zs542x79sh
```

```

**Comment by ludovicc** [comment]
I use https for all urls as suggested.

**Review by jonringer — COMMENTED** [review]
regression in `jd-gui` package.

also, please squash fixup commits to adhere to CONTRIBUTING.md

```
https://github.com/NixOS/nixpkgs/pull/92103
1 package failed to build:
jd-gui

4 packages built:
frostwire gradle jadx mxisd
```

**Comment by ludovicc** [comment]
jd-gui should work better, I pinned gradle to version 5 as used by this package.
squashed the commits.

File: pkgs/tools/security/jd-gui/default.nix [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, fetchFromGitHub, jre, jdk, gradle, makeDesktopItem, perl, writeText, runtimeShell }:
+{ stdenv, fetchFromGitHub, jre, jdk, gradle_5, makeDesktopItem, perl, writeText, runtimeShell }:
```
Comment (jonringer) [inline pkgs/tools/security/jd-gui/default.nix:1]: the gradle_5 pinning should probably be done in another commit

Comment (jonringer) [reply pkgs/tools/security/jd-gui/default.nix:1]:
```
jd-gui: pin gradle to 5.X
gradle: 5.6.4 -> 6.5.1 
```

**Comment by ludovicc** [comment]
@jonringer I have moved the gradle_5 pinning to #92292, this pr is now dependant of #92292

**Review by jonringer — APPROVED** [review]
```
https://github.com/NixOS/nixpkgs/pull/92103
4 packages built:
frostwire gradle jadx mxisd
```
